notebooks/Interpretability - ICE explainer.ipynb

```diff
@@ -0,0 +1 @@
ualizes the average dependence. Red line - PDP plot, black lines - ICE plots"]},{"cell_type":"code","execution_count":null,"metadata":{"application/vnd.databricks.v1+cell":{"inputWidgets":{},"nuid":"36bfc955-b6c8-4e9d-a28e-42c7e9889692","showTitle":false,"title":""}},"outputs":[],"source":["col_name =  'age_dependence'\n","overlay_ice_with_pdp(age_dep, df_education_num, col=col_name, figsize=(30, 10))"]},{"cell_type":"markdown","metadata":{"application/vnd.databricks.v1+cell":{"inputWidgets":{},"nuid":"d0a80081-a3f6-41d0-8f6d-b686220dbeaa","showTitle":false,"title":""}},"source":["#### Example 2: Categorical feature: \"occupation\""]},{"cell_type":"code","execution_count":null,"metadata":{"application/vnd.databricks.v1+cell":{"inputWidgets":{},"nuid":"1fee7eb9-6e89-480c-b864-fed2cfbf33f5","showTitle":false,"title":""}},"outputs":[],"source":["col_name = 'occupation_dependence'\n","occupation_dep = get_pandas_df_from_column(output, col_name)\n","\n","\n","plot_ice_categorical(occupation_dep, col_name, figsize=(30, 10))"]},{"cell_type":"markdown","metadata":{"application/vnd.databricks.v1+cell":{"inputWidgets":{},"nuid":"4f6494ac-6bf5-408c-9ac2-c44b54afba38","showTitle":false,"title":""}},"source":["### Conclusions\n","\n","\n","**Advantages:**\n","\n","1) Plots are intuitive to understand. One line represents the predictions for one instance if we vary the feature of interest.\n","\n","2) ICE curves can uncover more complex relationships.\n","\n","**Disadvantages:**\n","\n","1) ICE curves can only display one feature meaningfully - otherwise you should overlay multiple surfaces.\n","\n","2) Some points in the lines might be invalid data points according to the joint feature distribution. It causes by correlations between features.\n","\n","3) In ICE plots it might not be easy to see the average."]},{"cell_type":"markdown","metadata":{"application/vnd.databricks.v1+cell":{"inputWidgets":{},"nuid":"b974545e-371b-41b0-b9a4-adb3036d9eb8","showTitle":false,"title":""}},"source":["## Summary\n","\n","Partial dependence plots (PDP) and Individual Conditional Expectation (ICE) plots can be used to visualize and analyze interaction between the target response and a set of input features of interest.\n","\n","Both PDPs and ICEs assume that the input features of interest are independent from the complement features, and this assumption is often violated in practice.\n","\n","ICE shows the dependence on average, but if you want to observe features individually - you can use ICE.\n","\n","Using examples above we showed how it can be usefull to draw such plots to analyze how machine learning model made their predictions, what was important and how we can interpret the results."]}],"metadata":{"application/vnd.databricks.v1+notebook":{"dashboards":[],"language":"python","notebookMetadata":{"pythonIndentUnit":2},"notebookName":"PDP-ICE-tutorial-new","notebookOrigID":2416290700869370,"widgets":{}},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}
```

Comment:
nit: Perhaps pump this through a json formatter so the diff lines doing forward will be nice

Reply:
Can't find how to do it :( But you can hit the "View file" and it beautifully displays the file. The rest comments I fixed